packages/react/lib/Element/index.tsx

```diff
@@ -1,23 +1,56 @@
-import { Fragment, useMemo, useRef, useState } from 'react';
-import { Draggable, Droppable, Tooltip, classNames } from '@junipero/react';
+import {
+  type MouseEvent,
+  type MutableRefObject,
+  type Ref,
+  Fragment,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import type {
+  ComponentObject,
+  ComponentOverride,
+  ComponentOverrideObject,
+  ElementObject,
+  FieldOverride,
+} from '@oakjs/core';
+import {
+  Draggable,
+  Droppable,
+  Tooltip,
+  classNames,
+} from '@junipero/react';
 
 import { copyToClipboard } from '../utils';
-import { useBuilder } from '../hooks';
+import DisplayableSettings from '../DisplayableSettings';
+import Editable from '../Editable';
 import Icon from '../Icon';
-import Text from '../Text';
 import Option from '../Option';
-import Editable from '../Editable';
-import DisplayableSettings from '../DisplayableSettings';
+import Text from '../Text';
+import { useBuilder } from '../hooks';
+
+interface ElementProps {
+  element?: ElementObject;
+  parent?: ElementObject[];
+  parentComponent?: ComponentObject;
+  className?: string;
+  depth?: number;
+}
+
+interface editableRefObject {
+  current: MutableRefObject<any>;
+  toggle: () => void;
+}
 
 const Element = ({
   element,
   parent,
   parentComponent,
   className,
   depth = 0,
-}) => {
-  const innerRef = useRef();
-  const editableRef = useRef();
+}: ElementProps) => {
+  const innerRef = useRef<Ref<HTMLElement>>();
```

Comment:
looks weird, i think you wanted to do 
`const innerRef = useRef<HTMLElement>();`